StringCalculator.Core/Processor/StringCalculatorProcessor.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Linq;
+
+namespace StringCalculator.Core.Processor
+{
+    public  class StringCalculatorProcessor
+    {
+        public StringCalculatorProcessor()
+        {
+        }
+
+        public  int Add(string numbers)
+        {
+            if (numbers.Equals(""))
+                return 0;
+
+            var separator = FindSeparator(ref numbers);
+
+            string[] numbersArray = numbers.Split(separator, '\n');
+
+            CheckForNegativeNumbers(numbersArray);
+
+            var sum = 0;
+            foreach (var numberString in numbersArray)
+            {
+                var value = int.Parse(numberString);
+                if (value <= 1000)
+                {
+                    sum += value;
+                }
+            }
```

Comment:
Could you try change this loop using Linq?
Hint: use Select,Where,Sum functions